packages/node/src/node/ServerNode.ts

```diff
@@ -180,13 +206,104 @@ export class ServerNode<T extends ServerNode.RootEndpoint = ServerNode.RootEndpo
     protected override async initialize() {
         await ServerEnvironment.initialize(this);
 
+        this.env.get(FabricManager).events.deleted.on(fabric => this.sanitizeFabricData(fabric));
+
         await super.initialize();
     }
 
     /**
-     * By default on factory reset we erase all stored data.
+     * Sanitize Fabric scoped data and events when a fabric is removed.
+     * The logic walks through all endpoints and removes relevant fabric-scoped attribute values for the relevant fabric
+     * from the state. After all state changes are processed, it removes all occurrences of fabric-scoped events.
+     */
+    protected async sanitizeFabricData(fabric: Fabric) {
```

Comment:
A more robust process might be to create a set of valid fabric then delete objects referencing fabrics that aren't in the list.

That would allow us to eventually remove stale data even if e.g. the node crashes after the fabric is removed but before cleanup completed